tensorflow/core/kernels/BUILD

```diff
@@ -1590,6 +1590,17 @@ cc_header_only_library(
     deps = [":lookup"],
 )
 
+cc_library(
```

Comment:
This new library shouldn't be necessary. The `dataset.h` header is now in `tensorflow/core/framework` and can be included via the `//tensorflow/core:framework_headers_lib` target.

You might want to check out the targets in `//tensorflow/contrib/data/...` that build dataset implementations as custom op libraries. This has changed in the last month or so, as we've refactored things to make separate compilation possible for dataset implementations.